library/src/main/java/com/novoda/downloadmanager/MigrationExtractor.java

```diff
@@ -0,0 +1,59 @@
+package com.novoda.downloadmanager;
+
+import android.database.Cursor;
+
+import com.novoda.notils.logger.simple.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class MigrationExtractor {
+
+    private static final String BATCHES_QUERY = "SELECT batches._id, batches.batch_title FROM batches INNER JOIN DownloadsByBatch ON " +
+            "DownloadsByBatch.batch_id = batches._id WHERE DownloadsByBatch.batch_total_bytes = DownloadsByBatch.batch_current_bytes";
+    private final DatabaseWrapper database;
+
+    MigrationExtractor(DatabaseWrapper database) {
+        this.database = database;
+    }
+
+    List<Migration> extractMigrations() {
+        Cursor batchesCursor = database.rawQuery(BATCHES_QUERY);
+
+        List<Migration> migrations = new ArrayList<>();
+        while (batchesCursor.moveToNext()) {
+
+            String query = "SELECT uri, _data, total_bytes FROM Downloads WHERE batch_id = ?";
+            String batchId = batchesCursor.getString(0);
```

Comment:
We could extract the `0`, `1` as a constants

Reply:
93ef127e02c9bc9e18347e9669fb515a92fc5bc2